lib/src/cancelable_operation.dart

```diff
@@ -155,15 +158,14 @@ class CancelableCompleter<T> {
   ///
   /// [onCancel] will be called synchronously when the operation is canceled.
   /// It's guaranteed to only be called once.
-  CancelableCompleter({FutureOr Function() onCancel})
+  CancelableCompleter({FutureOr Function()? onCancel})
       : _onCancel = onCancel,
         _inner = Completer<T>() {
-    _operation = CancelableOperation<T>._(this);
+    operation = CancelableOperation<T>._(this);
   }
 
   /// The operation controlled by this completer.
-  CancelableOperation<T> get operation => _operation;
-  CancelableOperation<T> _operation;
+  late final CancelableOperation<T> operation;
```

Comment:
I think this gives the appearance of a setter. There won't be a static error for trying to set it... Is that a bug in the SDK?

Reply:
oh interesting...... cc @leafpetersen

Reply:
Going to leave this for now and follow up if we decide it needs to change.

Reply:
I do think this needs to change. https://github.com/dart-lang/async/pull/139